### PR TITLE
ambient: make waypoints handle dual stack services

### PR DESCRIPTION
**Pull request by howardjohn**
1. Make waypoint use all service addresses.
2. Make ServiceEntry with multiple addresses one service instead of N services. 

(2) has potential impacts beyond ambient so need to closely evaluate that

### CODE REVIEW

File: pilot/pkg/model/service.go [hunk 1]
```diff
@@ -1287,6 +1287,20 @@ func (s *Service) GetAllAddressesForProxy(node *Proxy) []string {
 	return nil
 }
 
+// UnconditionallyGetAllAddressesForProxy returns a k8s service's extra addresses to the cluster where the node resides.
```
Comment (howardjohn) [inline pilot/pkg/model/service.go:1290]: This duplication kind of sucks, intent was to make it so ambient is always dualstack even without features.EnableDualStack. TBD

**Comment by leosarra** [comment]
I'm in favor of having a single ServiceEntry with all addresses into it (similar to what K8s does, main address + addresses list) because
I feel that having multiple ServiceEntry (each one representing one address) is kinda altering the semantics of the object already.

I think these changes would be beneficial for all users, including non-ambient users. Do you expect some behavioral changes for the users (excluding possible buggy interactions with existing code)?

**Comment by howardjohn** [comment]
obsolete by https://github.com/istio/istio/pull/51939. Thanks @jewertow

**Pull request closed without merging**